### PR TITLE
AP-1998 integrate backend proceedings type search with frontend

### PR DESCRIPTION
**Pull request by RoseSAK**
## Integrate backend proceedings type search with frontend

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1998)

Refactored JS to send Ajax call to backend proceedings type full text search
@Alex-Swann refactored ProceedingsTypeFullTextSearch to work with substrings and with multiple search terms 
Converted jQuery to vanilla JS [See master ticket here] (https://dsdmoj.atlassian.net/browse/AP-1821) and removed second proceeding types JS file
Updated rspec tests for ProceedingsTypeFullTextSearch 
Updated Jest tests 

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/webpack/src/modules/proceedings.js [hunk 1]
```diff
@@ -1,126 +1,122 @@
...
-var typingTimer;                //timer identifier
-var doneTypingInterval = 1500;  //time in ms, 1.5 second for example
-var ariaText;
-
-async function getAll(){
-  const response = await axios.get('/v1/proceeding_types')
-  return response.data
+import axios from 'axios';
+import { hide, show, pluralize } from '../helpers';
+const doneTypingInterval = 500; // time in ms, decided on 400ms to make search seem instant (but it's not!)
```
Comment (mpw5) [inline app/webpack/src/modules/proceedings.js:3]: Really nitpicky, sorry, but the comment refers to 400ms when the actual interval is set as 500. Can we make them consistent?

File: app/webpack/src/modules/proceedings.js [hunk 2]
```diff
@@ -1,126 +1,122 @@
...
+  if (results.length > 0) {
+    const codes = results.map(obj => obj.code);
+    codes.forEach(code => {
+      // const element = $('#' + code)
+      const element = document.getElementById(code);
+
+      // We want to highlight anything text in <main> or <span> tags that
+      // matches the user's search criteria
+      const span = element.querySelector('span');
+      // remove h3 when the multiple proceedings feature flag is removed
```
Comment (mpw5) [inline app/webpack/src/modules/proceedings.js:72]: Might be worth adding a `TODO` to this comment to make it clearer that future action is required (similar to lines 43 and 109)

**Review by mpw5 — APPROVED** [review]
Looks good to me! A couple of nitpicky comments but no blockers 💯

File: app/webpack/src/modules/proceedings.js (hunk 1 above)
Comment (RoseSAK) [reply app/webpack/src/modules/proceedings.js:3]: oops sorry I changed it but forgot to change the comment, will update now!

File: app/webpack/src/modules/proceedings.js (hunk 2 above)
Comment (RoseSAK) [reply app/webpack/src/modules/proceedings.js:72]: will do!

**Pull request merged**